base/linalg/triangular.jl

```diff
@@ -319,7 +343,7 @@ end
 function A_mul_B!(A::UpperTriangular, B::StridedVecOrMat)
     m, n = size(B, 1), size(B, 2)
     if m != size(A, 1)
-        throw(DimensionMismatch("left and right hand side does not fit"))
+        throw(DimensionMismatch("B needs first dimension of size $(size(A,1)), has size $m"))
```

Comment:
I'd normally agree with you, however, in this case, A and B are part of the name of the function `A_mul_B`.  Maybe @kshyatt could put `B (right hand side) needs...`